soroban-auth/src/testutils.rs

```diff
@@ -0,0 +1,72 @@
+#![cfg(feature = "testutils")]
+#![cfg_attr(feature = "docs", doc(cfg(feature = "testutils")))]
+
+//! Utilities intended for use when testing contracts that use
+//! [`soroban_auth`](crate).
+
+pub mod ed25519 {
```

Comment:
This module would be good for the `to_ed25519` helper we use in most tests.

Reply:
Looks like it was called `make_identifier` in the auth tests, but was removed. What do you think about adding it back in and exposing it?

Reply:
Do you think we need the helper still? The `to_ed25519` helper takes a `ed25519_daley::Keypair` and created an `Identifier` for it. The utilities in this module provide key generation that already produces the needed `Identifier`.

```rust
pub fn to_ed25519(e: &Env, kp: &Keypair) -> Identifier {
    Identifier::Ed25519(kp.public.to_bytes().into_val(e))
}
```

Reply:
👇🏻 
https://github.com/stellar/rs-soroban-sdk/blob/90c89c036b51cbd152a3ab7da6e61243d6f3f02d/soroban-auth/src/testutils.rs#L20-L31

Reply:
Actually, I'm going to try making a change to this that will get you that same fn, but retain the way that the identifier is handed to the developer precreated. Commit incoming.

Reply:
I've added a function to `Keypair` for getting the `Identifier` for that type. That simplified the `Sign` functions argument inputs, but if you were wanting to get an `Identifier` for a `Keypair` in tests for other purposes you could use it for that too.

Reply:
Yeah that works. This can be used in our tests like [here](https://github.com/stellar/soroban-examples/blob/58cf2648d130bf82396d87bead559a32e08b3433/liquidity_pool/src/test.rs#L67) for example. Thanks.

Reply:
Will downstream dependencies have to alias `Identifier as IdentifierValue` as well?

Reply:
> Will downstream dependencies have to alias Identifier as IdentifierValue as well?

No I don't think so, not in my testing. If I merge this and in the process of updating examples discover I'm wrong about this, I'll circle back and rename the trait.

Reply:
> Yeah that works. This can be used in our tests like [here](https://github.com/stellar/soroban-examples/blob/58cf2648d130bf82396d87bead559a32e08b3433/liquidity_pool/src/test.rs#L67) for example. Thanks.

Note that doing so will be unnecessary, because `generate()` returns an `Identifier` and a signer.

Reply:
For example:
https://github.com/stellar/rs-soroban-sdk/blob/ebb802783213b402494048a060375dabc6eff0fe/soroban-auth/tests/test.rs#L74

Reply:
Ah yeah I missed that. Sounds good.

Reply:
I think the trait will still be helpful. You won't need to pass a keypair and an identifier around.